hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -2264,7 +2264,14 @@ public boolean isIgnoreNotFound(){
 	}
 
 	public boolean hasNotFoundAction() {
-		return notFoundAction != null;
+		if ( notFoundAction != null ) {
+			return true;
+		}
+		else if ( referencedPropertyName != null && cardinality != Cardinality.MANY_TO_ONE ) {
+			final AttributeMapping mappedByAttribute = entityMappingType.findAttributeMapping( referencedPropertyName );
+			return mappedByAttribute instanceof ToOneAttributeMapping && ( (ToOneAttributeMapping) mappedByAttribute ).hasNotFoundAction();
+		}
```

Comment:
Maybe I was a bit unclear on this, but what I meant is that a one-to-one association which uses `mappedBy` should behave the same way as if `@NotFound` is annotated. You don't have to check if the inverse attribute has a not found action, this is irrelevant.